persistence-api/src/main/java/io/stargate/db/Persistence.java

```diff
@@ -44,6 +44,8 @@
 
   ClientState newClientState(String name);
 
+  AuthenticatedUser<?> newAuthenticatedUser(String name);
```

Comment:
Opening this API does not sound right to me. Since we validate credentials (token) via `AuthenticationService` service and by-pass the negotiator from `Persistence` in this case, would it be preferable to have a local implementation of `AuthenticatedUser` (in the `cql` module) for this use case?

Reply:
Having thought a bit more about this, I guess the `AuthenticationService` should perhaps negotiate with `Persistence` and establish an authenticated `ClientState`... WDYT?

Reply:
The problem with this is you end up duplicating the logic over and over in all the different persistence implementation.

Reply:
Let me take a look a this a bit more.

Reply:
The issue here I need to interoperate with an existing `ClientState`:
https://github.com/stargate/stargate/blob/v0.0.8/cql/src/main/java/org/apache/cassandra/stargate/transport/internal/messages/AuthResponse.java#L78-L79

Also, I would prefer not refactor `AuthReponse` as part of this feature.

Reply:
Understood, but in principle, having an API that just creates an authenticated user looks dangerous to me. In the long run, would it make sense to direct all authentication call through the `AuthenticationService` only? `Persistence` implementations could connect to it through a separate API following the model similar to JRE's `javax.security.auth.spi.LoginModule`, WDYT?

Reply:
Would it be worth opening a follow-up ticket?

Reply:
> Understood, but in principle, having an API that just creates an authenticated user looks dangerous to me